packages/app/src/cli/models/app/app.ts

```diff
@@ -7,9 +7,13 @@ import {fileRealPath, findPathUp} from '@shopify/cli-kit/node/fs'
 import {joinPath, dirname} from '@shopify/cli-kit/node/path'
 
 export const AppConfigurationSchema = zod.object({
-  scopes: zod.string().default(''),
   extension_directories: zod.array(zod.string()).optional(),
   web_directories: zod.array(zod.string()).optional(),
+  name: zod.string().optional(),
```

Comment:
I'm not sure on this -- if I load an App current-style then these fields all always be undefined. And if I load new-style then these fields should all always have some value (I think?)

This is tricky as we're dealing with two quite distinct models that underpin such a lot. But if we get the approach wrong we'll feel the debt on this for a very long time. So its a small thing but might have an outsized impact. When we're working new-style for instance, we don't want to drop `if (app.client_id !== undefined)` everywhere when it reality it'd always be set.

I wonder if this might be better done as a [tagged/discriminated union](https://zod.dev/?id=discriminated-unions). If (for instance) we used `name` as the discriminator then:
- if `name` is unset, it can be treated old-style
- if `name` is set, it can be treated new-style, with validation & guarantees on these new fields (and the new sections etc.)

Another alternative could be to have the apploader return something thats an extension of the current output when you're loading new-style.